### ISSUE TITLE
v1.0.5 Exception

### ISSUE THREAD

**Issue by barrycarr** [koxudaxi/poetry-pycharm-plugin#175]
**Describe the bug**
v1.0.5 Kills the run/debug configuration pane in Intellij

**To Reproduce**
1. Go to 'Run -> Edit Configurations...'
2. Click on an existing python configuration
3. Configuration Pane on the right handside of the window is blank
4. Click on the "+" button, click on the Python option - nothing happens 

**Expected behaviour**
I should be able to edit an existing python run/debug configs or, create new ones

**Screenshots**
![image](https://user-images.githubusercontent.com/280746/107244738-442c7d80-6a26-11eb-839d-f91bd0890198.png)

**Environments (please complete the following information):**
 - IDE: Intellij Ultimate 2020.3.2
 - OS: Kubuntu 20.04
 - Poetry Version : 1.1.4
 - Plugin version 1.0.5

**Additional context**
The IDE reported the following exception from the plugin:
```
Stacktrace
com.intellij.diagnostic.PluginException: Cannot create class com.koxudaxi.poetry.PyProjectTomlWatcher [Plugin: com.koxudaxi.poetry]
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:725)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.instantiateClass(ExtensionComponentAdapter.java:51)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter$SimpleConstructorInjectionAdapter.instantiateClass(XmlExtensionAdapter.java:133)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.createInstance(ExtensionComponentAdapter.java:43)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.createInstance(XmlExtensionAdapter.java:69)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapter(ExtensionPointImpl.java:472)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.access$200(ExtensionPointImpl.java:37)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl$1.next(ExtensionPointImpl.java:367)
	at com.intellij.openapi.extensions.impl.ExtensionProcessingHelper.forEachExtensionSafe(ExtensionProcessingHelper.java:27)
	at com.intellij.openapi.extensions.ExtensionPointName.forEachExtensionSafe(ExtensionPointName.java:49)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:203)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:157)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorComponent.createEditor(TextEditorComponent.java:133)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorComponent.<init>(TextEditorComponent.java:79)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorImpl$PsiAwareTextEditorComponent.<init>(PsiAwareTextEditorImpl.java:107)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorImpl.createEditorComponent(PsiAwareTextEditorImpl.java:85)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorImpl.<init>(TextEditorImpl.java:52)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorImpl.<init>(PsiAwareTextEditorImpl.java:36)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorProvider.createEditor(PsiAwareTextEditorProvider.java:31)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openFileImpl4Edt(FileEditorManagerImpl.java:891)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$openFileImpl4$17(FileEditorManagerImpl.java:860)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.runBulkTabChange(FileEditorManagerImpl.java:1680)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$openFileImpl4$18(FileEditorManagerImpl.java:859)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:216)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:24)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:199)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeAndWait$8(ApplicationImpl.java:476)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:126)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:85)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:134)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:47)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:190)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:976)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:454)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:453)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:501)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:684)
	... 52 more
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.koxudaxi.poetry.PoetryKt
	at com.koxudaxi.poetry.PyProjectTomlWatcher.<init>(poetry.kt:430)
	... 57 more
```

**Comment by cos18** [comment]
I also have similar issue when initialize project in PyCharm. I'm working in Pycharm 2020.3.3 + M1 mac mini with macOS 11.2

```
java.lang.NoClassDefFoundError: org/toml/lang/psi/TomlTableHeader
	at com.koxudaxi.poetry.PoetryKt.tomlTableHeaderHasKey(poetry.kt:721)
	at com.koxudaxi.poetry.PoetryKt.<clinit>(poetry.kt:722)
	at com.koxudaxi.poetry.PoetrySdkProvider.getSdkAdditionalText(PoetrySdkProvider.kt:59)
	at com.jetbrains.python.sdk.PySdkRenderingKt.name(PySdkRendering.kt:31)
	at com.jetbrains.python.sdk.PySdkRenderingKt.name(PySdkRendering.kt:18)
	at com.jetbrains.python.sdk.PySdkPopupFactory$Companion.nameInPopup(PySdkPopupFactory.kt:29)
	at com.jetbrains.python.sdk.PySdkPopupFactory$Companion.descriptionInPopup(PySdkPopupFactory.kt:36)
	at com.jetbrains.python.configuration.PySdkStatusBar.getWidgetState(PySdkStatusBar.kt:75)
	at com.intellij.openapi.wm.impl.status.EditorBasedStatusBarPopup.lambda$update$2(EditorBasedStatusBarPopup.java:266)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:376)
	at com.intellij.util.Alarm$Request.run(Alarm.java:362)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:220)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:201)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:324)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:85)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:134)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:47)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:190)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:976)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:454)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:453)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:501)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.ClassNotFoundException: org.toml.lang.psi.TomlTableHeader PluginClassLoader[PluginDescriptor(name=Poetry, id=com.koxudaxi.poetry, path=/Users/sunpark/Library/Application Support/JetBrains/PyCharm2020.3/plugins/poetry-pycharm-plugin, version=1.0.5)] com.intellij.ide.plugins.cl.PluginClassLoader@64d2e675
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:129)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 42 more
```

**Comment by koxudaxi** [comment]
@barrycarr 
Thank you for creating this issue.
I'm sorry I can't reproduce your error.
Did you have more hints?

> 2. Click on an existing python configuration

For example, Does the existing python configuration use poetry?
Also, Does the error happen again?

@cos18
Thank you for posting the error.
I guess the reason is not the same.
I might break the dependency on the plugin in the last commit.
Would you install this [toml plugin](https://plugins.jetbrains.com/plugin/8195-toml)?

**Comment by incoggnito** [comment]
Installing the `toml plugin` solves the issue for me!

**Comment by koxudaxi** [comment]
@incoggnito 
Thank you for your report.
Also, I pushed version 1.0.6 that is a fixed version.
The version will be published while two business days.

**Comment by barrycarr** [comment]
@koxudaxi Hi, My issue occurred on run configuration for a python project that wasn't using poetry. We are in the process of moving over to poetry and I was in the process of creating a new poetry based run config when I discovered the issue. This was the first time I'd attempted to use your poetry plugin.

**Comment by koxudaxi** [comment]
@barrycarr 
Thank you for posting the detail.
I watched your log. The error was happened by running `poetry --version` to check the poetry version.
This plugin calls an external command on a 10-sec timeout.
But, poetry often is a slow response. The plugin might kill the `poetry` command by timeout.
I have changed the timeout duration from 10-sec to 30-sec in version `1.0.6`.
If you always get the same error then, the reason probably is another one and I will investigate this problem again.
